crates/proof-of-sql/src/sql/proof_plans/union_exec_test.rs

```diff
@@ -0,0 +1,221 @@
+use super::test_utility::*;
+use crate::{
+    base::{
+        database::{
+            owned_table_utility::*, table_utility::*, ColumnType, OwnedTable,
+            OwnedTableTestAccessor, TableTestAccessor, TestAccessor,
+        },
+        map::indexmap,
+        scalar::Curve25519Scalar,
+    },
+    sql::{
+        proof::{
+            exercise_verification, FirstRoundBuilder, ProvableQueryResult, ProverEvaluate,
+            VerifiableQueryResult,
+        },
+        proof_exprs::test_utility::*,
+    },
+};
+use blitzar::proof::InnerProductProof;
+use bumpalo::Bump;
+
+#[test]
+fn we_can_prove_and_get_the_correct_empty_result_from_a_union_exec() {
```

Comment:
Can you add some more edge cases:
* Union of 1 table.
* Union of 0 tables.

Reply:
@JayWhite2357 `UnionExec` only exists when there is at least one `UNION` in a query which requires at least two tables. Hence we shouldn't do more than an `assert` which I already did to ban unions with 0 or 1 tables.